src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx

```diff
@@ -31,54 +31,45 @@ export const useStakingAction = () => {
     if (wallet) {
       try {
         const adapter = chainAdapterManager.byChain(data.asset.chain)
-        const adapterType = adapter.getType()
 
         let result
 
         const { chainSpecific, validator, action, value } = data
         const memo = validator === SHAPESHIFT_VALIDATOR_ADDRESS ? 'Delegated with ShapeShift' : ''
 
-        if (adapterType === ChainTypes.Cosmos) {
-          switch (action) {
-            case StakingAction.Claim: {
-              result = await (
-                adapter as cosmossdk.cosmos.ChainAdapter
-              ).buildClaimRewardsTransaction({
-                wallet,
-                validator,
-                chainSpecific,
-                memo,
-              })
-              break
-            }
-            case StakingAction.Stake: {
-              result = await (adapter as cosmossdk.cosmos.ChainAdapter).buildDelegateTransaction({
-                wallet,
-                validator,
-                value,
-                chainSpecific,
-                memo,
-              })
-              break
-            }
-            case StakingAction.Unstake: {
-              result = await (adapter as cosmossdk.cosmos.ChainAdapter).buildUndelegateTransaction({
-                wallet,
-                validator,
-                value,
-                chainSpecific,
-                memo,
-              })
-              break
-            }
-            default: {
-              break
-            }
+        switch (action) {
+          case StakingAction.Claim: {
+            result = await (adapter as cosmossdk.cosmos.ChainAdapter).buildClaimRewardsTransaction({
```

Comment:
you can define this adapter once to avoid casting three times

you can also define the payload once, and just use this switch statement to spread it onto actions

will eliminate most of these lines here